cmd/main.go

```diff
@@ -115,21 +130,54 @@ func main() {
 
 					pubURLstr := cCtx.String("public-url")
 					if pubURLstr == "" {
-						pubURLstr = fmt.Sprintf("http://localhost:%d", cCtx.Int("port"))
-						log.Errorf("Public URL is not configured, using: %s", pubURLstr)
+						if os.Getenv("PUBLIC_URL") != "" {
+							pubURLstr = os.Getenv("PUBLIC_URL")
+						} else {
+							pubURLstr = fmt.Sprintf("http://localhost:%d", cCtx.Int("port"))
+							log.Errorf("Public URL is not configured, using: %s", pubURLstr)
+						}
 					}
 					pubURL, err := url.Parse(pubURLstr)
 					if err != nil {
 						return fmt.Errorf("parsing public URL: %w", err)
 					}
 
+					announceURL := *publisher.AnnounceURL
+					if os.Getenv("ANNOUNCE_URL") != "" {
```

Comment:
any reason this one and the next two appear to be env only? I would just make them CLI flags too that can read from env (but can also be set at startup)